src/Xamarin.Android.Build.Tasks/Utilities/MonoAndroidHelper.cs

```diff
@@ -692,5 +679,38 @@ public static string FixUpAndroidResourcePath (string file, string resourceDirec
 			}
 			return string.Empty;
 		}
+
+		/// <summary>
+		/// Converts .NET 5 RIDs to Android ABIs or an empty string if no match.
+		/// 
+		/// Known RIDs:
+		/// "android.21-arm64" -> "arm64-v8a"
+		/// "android.21-arm"   -> "armeabi-v7a"
+		/// "android.21-x86"   -> "x86"
+		/// "android.21-x64"   -> "x86_64"
+		/// "android-arm64"    -> "arm64-v8a"
+		/// "android-arm"      -> "armeabi-v7a"
+		/// "android-x86"      -> "x86"
+		/// "android-x64"      -> "x86_64"
+		/// </summary>
+		public static string RuntimeIdentifierToAbi (string runtimeIdentifier)
+		{
+			if (string.IsNullOrEmpty (runtimeIdentifier)) {
+				return "";
+			}
+			if (runtimeIdentifier.EndsWith ("-arm64", StringComparison.OrdinalIgnoreCase)) {
+				return "arm64-v8a";
+			}
+			if (runtimeIdentifier.EndsWith ("-arm", StringComparison.OrdinalIgnoreCase)) {
+				return "armeabi-v7a";
+			}
+			if (runtimeIdentifier.EndsWith ("-x86", StringComparison.OrdinalIgnoreCase)) {
+				return "x86";
+			}
+			if (runtimeIdentifier.EndsWith ("-x64", StringComparison.OrdinalIgnoreCase)) {
+				return "x64";
```

Comment:
Shouldn't this be `x86_64`, as per the comment on line 694?

Reply:
Yeah, this is wrong 🤦‍♂️ . I updated the test so it will try every architecture & RID.